even more recycling

The4codeblocks opened the pull request.

The4codeblocks committed:
even more recycling

The4codeblocks committed:
Update crafts.lua

The4codeblocks committed:
missing commas

The4codeblocks committed:
missed comma

The4codeblocks committed:
Update crafts.lua

The4codeblocks committed:
shutting up luacheck

The4codeblocks committed:
Update crafts.lua

The4codeblocks committed:
add dependency

The4codeblocks committed:
Merge branch 'pandorabox-io:master' into patch-3

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
 -- format is {recipe, output[, replacement]}
 
-local cookable_items = {
+local cookable_items = { --DRAWER UPGRADE, BRIDGER
```
This all caps comment doesn't make any sense to me

The4codeblocks replied:
oh I forgot to clean those up
those are plans

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
+		{ "signs_road:signs_road:green_sign", "default:steel_ingot 1" },
+		{ "signs_road:signs_road:blue_sign", "default:steel_ingot 1" },
+		{ "signs_road:signs_road:yellow_sign", "default:steel_ingot 1" },
+		{ "signs_road:signs_road:white_sign", "default:steel_ingot 1" },
```
indentation error

The4codeblocks replied:
that's intentional
it's meant to be a category thing

SwissalpS replied:
well please don't do that with indents.
It is so easy and clean to make categories with comments and empty lines.
```lua
local t = {
    -- Category A
    {.....},
    {.....},

    -- Category B
    {.....},
    {.....},

    -- Category N
    {.....},
    {.....},
}
```

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
+technic.register_compressor_recipe({ input = {"charcoal:charcoal_lump 80"}, output = "default:coal_lump 1" })
+
+technic.register_compressor_recipe({ input = {"factory_bridges:i00 30"}, output = "factory_bridges:b00 1" })
+technic.register_grinder_recipe({ input = {"factory_bridges:b00 10"}, output = "technic:wrought_iron_dust 3" })
```
you have nice space after '{' and before '}' everywhere else, why stop here?

SwissalpS replied:
also this needs to be wrapped in a clause checking that technic actually is being used.

SwissalpS commented:
quite a lot of items to check, this can take a while ;)

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
 
+-- the density of ethereal mod charcoal is ~1/10 of coal, otherwise it's pure carbon
+-- the volume of charcoal mod charcoal is 1/8 of ethereal mod charcoal, since it uses planks for 2x the output count
+technic.register_compressor_recipe({ input = {"charcoal:charcoal_lump 80"}, output = "default:coal_lump 1" })
```
If I remember correctly this had been discussed somewhere and it was decided that we didn't want to be able to convert wood into diamonds.

SwissalpS replied:
Maybe we can discuss turning charcoal into carbon-plates, idk though.

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
+	{ "spacesuit:helmet", "default:steel_ingot 14" },
+	{ "spacesuit:pants", "default:steel_ingot 4" },
+	{ "spacesuit:boots", "default:steel_ingot 15" },
+	{ "signs_road:blue_street_sign", "default:steel_ingot 1" },      -- floor(6 in / 4 out) = 1
```
why isn't this sign with the other signs?

SwissalpS commented on crafts.lua:
```diff
@@ -110,25 +110,103 @@ core.register_craft({
+		{ "signs_road:signs_road:white_sign", "default:steel_ingot 1" },
+	{ "more_chests:toolbox_steel", "default:steel_ingot 8" },        -- pickaxe of unknown material
+	{ "bike:bike", "default:steel_ingot 7" },
+	{ "signs_road:large_street_sign", "default:steel_ingot 6" },
```
signs bundled with signs makes sense to me, especially since there doesn't seem to be an alphabetical order to your additions. (it wasn't perfect before either)

The4codeblocks committed:
tidy up recipes a bit

"Refactor technic recipes with global check" - copilot
not a complete description

SwissalpS commented on crafts.lua:
```diff
@@ -111,24 +111,114 @@ core.register_craft({
 })
 
 
+if core.global_exists("technic") then
```
this isn't clean. please use ``core.get_modpath()``

SwissalpS commented:
Speaking of empty lines, please make sure your empty lines are actually empty ;)
See luacheck errors. (Any good code editing software will automatically do this for you once it is set up correctly.)

The4codeblocks committed:
replace global_exists with get_modpath, clean up indentation

"Refactor sign recipes for consistency" - copilot
very off

The4codeblocks committed:
shut luacheck up

SwissalpS commented:
https://github.com/pandorabox-io/in-game/issues/245 -> this will unlikely ever get merged as long as it contains a path from wood to diamonds

``technic.register_compressor_recipe({ input = {"charcoal:charcoal_lump 80"}, output = "default:coal_lump 1" })``

The4codeblocks committed:
comment out undesired recipe

1st copilot suggestion:
Comment out charcoal recipe in crafts.lua
Commented out the charcoal recipe registration for technic mod.

2nd suggestion:
Disable unwanted charcoal compressor recipe
Commented out undesired compressor recipe for charcoal.

The4codeblocks commented:
> [pandorabox-io/in-game#245](https://github.com/pandorabox-io/in-game/issues/245) -> this will unlikely ever get merged as long as it contains a path from wood to diamonds
> 
> `technic.register_compressor_recipe({ input = {"charcoal:charcoal_lump 80"}, output = "default:coal_lump 1" })`

done in https://github.com/pandorabox-io/pandorabox_custom/pull/93/commits/3f9d20a759b18122d6cbae7d55e58ac1d21c49b0

The4codeblocks commented:
so....?

The4codeblocks commented:
hello?

SwissalpS committed:
crafts.lua remove --[[ ... --]]

these comment blocks should only be used for temporary comment blocks or at top/bottom of file aka out of the way.
Otherwise they can complicate quick debugging sessions.

SwissalpS commented:
I've checked some of the maths and it seems to be OK.
The longer I look at this list though, the less pros I see.
Pros:
- Encentivize IRL recycling mentality.
- Buckets, battle-axes, yes, why not.

Cons:
- Adds more lag to crafting, something we try to avoid.
- Removes consequences, players should have to think about what and how much they craft.
- Screwdriver, trash-can and cutting board: absolutely not. Whoever crafts too many of those isn't thinking much or has reached 'creative' mode and won't miss some steel.

Originally the recycling recipes were added for items found in dungeon chests. Armour and weapons stretched that already quite a bit IMO. Most of the suggestions in this PR don't seem to make the game more interesting.

I vote no.

If @BuckarooBanzay , @S-S-X and @OgelGames think this PR adds value, fine. I'm not spending more time on it.
